### PR TITLE
Gen.weighted

### LINKED ISSUE #25: Make Gen.frequency Less Terrifying
We need a version that maps pure across the second part of each tuple of normal values rather than generators.

### PR DESCRIPTION
**Pull request by robrix**
- [x] Work around Swift linker bug.
- [x] Fixes #25.

### CODE REVIEW

**Comment by robrix** [comment]
Ready for review. Note that I’ve implemented `weighted` over `SequenceType` rather than `Array` simply as a workaround for rdar://21172325.

**Comment by CodaFi** [comment]
2 requests before I merge:

- Can you redefine `frequency` with `SequenceType` like you did here?
- See the TODO about linker errors?  Mind sticking your radar number in there?

File: SwiftCheck/Combinators.swift [hunk 1]
```diff
@@ -51,6 +51,12 @@ extension Gen {
 			return pick(l)(lst: xs)
 		}
 	}
+
+	/// Given a list of values and weights associated with them, this function randomly selects and
+	/// uses a Generator wrapping one of the values.
+	public static func weighted<S: SequenceType where S.Generator.Element == (Int, A)>(xs : S) -> Gen<A> {
```
Comment (CodaFi) [inline SwiftCheck/Combinators.swift]: Little nitpick

`S : SequenceType`

**Comment by robrix** [comment]
Where would you like `frequency` moved to…?

**Comment by robrix** [comment]
I’ve done the other bits now :+1: Just unclear on where you’d like `frequency` moved to.

**Comment by CodaFi** [comment]
I want it moved back to `Combinators.swift`  I ran into the same linker errors as you, I just gave up trying to solve it and left it in with the main declaration.

**Comment by CodaFi** [comment]
Nevermind.  Am crazy. 

:boat:

**Pull request merged**